### PR TITLE
Add helper image `merger` to merge yaml files

### PR DESCRIPTION
**Pull request by lukaszo**
## Description

Changes proposed in this pull request:

- Add helper image `merger` to merge yaml files

## Testing

Build image
```shell
docker build -t merger .
```

Create few yaml files:

`input.yaml`:
```yaml
a: 1
```

`additional:yaml`:
```yaml
b: 2
```

`extra.yaml`:
```yaml
c: 3
```

Run merger
```bash
docker run --mount src=`pwd`,target=/yamls,type=bind -e OUT=/yamls/all.yaml merger
```

Check the output file:
`all.yaml`
```yaml
input:
  a: 1
additional:
  b: 2
extra:
  c: 3
```
## Related issue(s)

#419

### CODE REVIEW

File: hack/images/merger/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# inputs-merger
+
+## Overview
+
+This folder contains Docker image with a helper for merging input parameters in manifests.
```
Comment (mszostok) [inline hack/images/merger/README.md:5]:
```suggestion
This folder contains the Docker image which merges multiple input YAML files into a single one.
```

File: hack/images/merger/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The Docker image consists a helper script, `merger.sh`. The script is an entrypoint of the image, and it is used to prefix and merge all yaml files found in `$SRC` directory.
```
Comment (mszostok) [inline hack/images/merger/README.md:7]:
```suggestion
The Docker image contains `merger.sh` helper script. The script is an entrypoint of the image, and it is used to prefix and merge all YAML files found in `$SRC` directory.
```

File: hack/images/merger/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+To build the Docker image, run this command:
+
+```bash
+docker build -t merger .
+```
+
+## Configuration
+
+You can configure the merger script passing the following environment variables:
```
Comment (mszostok) [inline hack/images/merger/README.md:20]:
```suggestion
You can configure the merger script by passing the following environment variables:
```

File: hack/images/merger/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+| Variable                  | Default      | Description                                      |
+| ------------------------- | ------------ | ------------------------------------------------ |
+| SRC                       | /yamls       | Path to the directory with yaml files.           |
+| OUT                       | /merged.yaml | Output file with prefixed and merged yaml files. |
```
Comment (mszostok) [inline hack/images/merger/README.md:25]:
```suggestion
| SRC                       | /yamls       | Path to the directory with YAML files.           |
| OUT                       | /merged.yaml | Output file with prefixed and merged YAML files. |
```

File: hack/images/merger/merger.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+SRC=${SRC:-"/yamls"}
+OUT=${OUT:-"/merged.yaml"}
+
+# prefix each file with its filename
+for filename in "${SRC}"/*.yaml; do
```
Comment (mszostok) [inline hack/images/merger/merger.sh:7]: WDYT about ignoring the extension? or maybe we should parametrize that? or simply support two most popular extensions, `yaml` and `yml`.

**Review by mszostok — COMMENTED** [review]
Tested and works good 👍 but please address small comments

- Please update the main README: https://github.com/capactio/capact/blob/926517cc92f3680868fadf2e2b1ab821b987859b/hack/images/README.md 
(you can also get rid of this structure)
- Please add this image also here: https://github.com/capactio/capact/blob/b3228c75c06b33f5cb1df6fdcbdf11e938338fe1/Makefile#L22

File: hack/images/merger/merger.sh (hunk 5 above)
Comment (lukaszo) [reply hack/images/merger/merger.sh]: let's ignore the extension

File: hack/images/README.md [hunk 6]
```diff
@@ -13,5 +13,6 @@ The structure of the folder looks as follows:
 ```
   ├── jinja2                 # Image containing a containerized jinja CLI tool assisting with rendering templatized OCF content
   ├── json-go-gen            # The image that contains quicktype to generates Go struct from JSON Schemas
-  └── graphql-schema-linter  # The image that contains graphql-schema-linter to lint GraphQL files
+  ├── graphql-schema-linter  # The image that contains graphql-schema-linter to lint GraphQL files
+  └── merger                 # The image that contains yaml merger helper scriptyaml merger helper script
```
Comment (mszostok) [inline hack/images/README.md:17]:
```suggestion
  └── merger                 # The image that contains YAML merger helper
```
hm.. sth was merged here 🤔 :stuck_out_tongue_winking_eye:

**Review by mszostok — APPROVED** [review]

File: hack/images/README.md (hunk 6 above)
Comment (lukaszo) [reply hack/images/README.md]: From what I see, during `argo retry` the pod is recreated(with the same name) and artifacts are downloaded again. We should be good here

**Pull request merged**